src/index.js

```diff
@@ -4,22 +4,85 @@
  * TOC Options
  *
  * @type {Object}
- * @prop {?String} options.after - tag, class, id
+ * @prop {?String} options.insert - insertion information with position key
+ *   ('after', 'before', 'afterChildren', 'beforeChildren') and selector value
+ *   ('tag', '.class', '#id')
  * @prop {?String} options.title - toc title
  * @prop {?Array[String, String, Boolean]} options.toggle - toggle button text options
  * @prop {?boolean} options.ignoreMissingSelector - don't throw an error if the
  *   selector is not found
  */
 module.exports = function (options = {}) {
   const {
-    after = 'h1',
+    insert = { after: 'h1' },
     title = 'Content',
     ignoreMissingSelector = false,
     toggle // ['show', 'hide', true],
   } = options
 
-  if (typeof after !== 'string') {
-    throw new PostHtmlTocError(`unexpected 'options.after': ${after}`)
+  if (typeof insert !== 'object') {
+    throw new PostHtmlTocError(`unexpected 'options.insert': ${insert}`)
+  }
+
+  const inserts = Object.entries(insert)
+
+  if (inserts.length !== 1) {
+    throw new PostHtmlTocError(`unexpected number of entries in 'options.insert': ${insert}`)
+  }
+
+  const [position, selector] = inserts[0]
```

Comment:
mb `insert: 'after h1'` `insert.split(' ')` ?

Reply:
I think it's better to avoid parsing strings more than necessary and to use syntax for some basic validation instead. If you'd prefer something less brace-y than `insert: { after: 'h1' }`, how about `insert: ['after', 'h1']`? If that doesn't work for you, I can live with `insert: 'after h1'`.

Reply:
I think either `{ after: 'h1' }` or `['after', 'h1']` is slightly better than `'after h1'`, but I'll go with whatever you choose.

Reply:
let's leave the original proposal